lib/bin/run_test.js

```diff
@@ -51,9 +51,13 @@ function invokeGeneratorOrPromiseFunction(fun) {
   }
 
   if (result && result.next) {  // This looks like a generator
-    return co(result);
+    return when.promise(function(resolve) {
+      // fixme: the promise returned by co seems to be
+      //        incompatible with the ones that when uses.
+      co(result).then(resolve);
```

Comment:
Also, if you decide to not use `return when(co(result));`, the right thing to do is to use `.done(resolve, reject)` instead of `.then(resolve)`, otherwise debugging tools will treat this as a promise that no one used the result of.